firebase-database/ktx/src/main/kotlin/com/google/firebase/database/ktx/Database.kt

```diff
@@ -59,6 +65,39 @@ inline fun <reified T> MutableData.getValue(): T? {
     return getValue(object : GenericTypeIndicator<T>() {})
 }
 
+/**
+ * Run a transaction on the data at this location.
+ * Returns the new data at the location.
+ *
+ * @param fireLocalEvents Defaults to true. If set to false, events will only be fired for the
+ *     final result state of the transaction, and not for any intermediate states
+ * @param handler An object to handle running the transaction
+ */
+suspend fun DatabaseReference.runTransaction(
+    fireLocalEvents: Boolean = true,
+    handler: (MutableData) -> Transaction.Result
+) = suspendCancellableCoroutine<DataSnapshot> { continuation ->
+    runTransaction(object : Transaction.Handler {
+        override fun doTransaction(currentData: MutableData): Transaction.Result {
+            return handler(currentData)
+        }
+
+        override fun onComplete(
+            error: DatabaseError?,
+            committed: Boolean,
+            currentData: DataSnapshot?
+        ) {
+            if (error != null) {
+                continuation.resumeWithException(error.toException())
+            } else {
+                if (currentData != null) {
+                    continuation.resume(currentData)
+                }
+            }
+        }
```

Comment:
I just realized I've ignored 2 things when implementing this:
- the `committed` boolean
- the case where both `error` and `currentData` are null: I assumed it would never happen, but it seems like it can as seen in this test:
    https://github.com/firebase/firebase-android-sdk/blob/fe17099d7a156e22faca77bef41b1b56a5116fd9/firebase-database/src/androidTest/java/com/google/firebase/database/TransactionTest.java#L150-L173

Reply:
Alternatives Considered:

1. Have this method return a [`Pair<DataSnapshot, Boolean>`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-pair/) to able to return both the committed boolean and the new data at location. I don't remember seeing any API return a Pair, but its usage isn't so different from destructure declarations which we already have in our SDK.
    ```kotlin
    try {
        val (snapshot, committed) = postRef.runTransaction { mutableData: MutableData ->
             // (actual transaction code omitted for brevity)
             return@runTransaction Transaction.success(mutableData)
        }
        // transaction succeeded and currentData has the current data at the location
    } catch (e: DatabaseException) {
        // handle errors
    }
    ```
1. Only return a `DataSnapshot` in cases where the transaction was successful. Aborted transactions will return a null `DataSnapshot` and won't throw an exception.

Reply:
After giving it some thought, I think Alternative 1 makes more sense as it doesn't change the current RTDB transaction behavior.

Alternative 2 can be consider as a behavior change if developers are migrating from the current callback to this new suspend function.